engine/src/main/java/org/terasology/network/BroadCastServer.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.network;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BroadCastServer {
+
+    private static final Logger logger = LoggerFactory.getLogger(BroadCastServer.class);
+    private static final String DISCOVERY_REQUEST= "DISCOVERY_REQUEST";
+    private static final String DISCOVERY_RESPONSE = "DISCOVERY_RESPONSE";
+
+    private static boolean turnOnBroadcast;
+
+    private DatagramSocket receiveSocket;
+    private DatagramSocket sendSocket;
+
+    public BroadCastServer(){
+        turnOnBroadcast=false;
+    }
+
+    public boolean isActive() {
+        return turnOnBroadcast;
+    }
+
+    public void setTurnOnBroadcast(boolean active) {
+        this.turnOnBroadcast=active;
+    }
+
+    public void startBroadcast() {
+        BroadCastServer.turnOnBroadcast = true;
+        try {
+            sendBroadCast();
+            Executors.newSingleThreadExecutor().execute(() ->  {
+                while (!Thread.currentThread().isInterrupted()) {
+                    try {
+                        while (turnOnBroadcast) {
+                            listenToBroadCast();
+                        }
+                    } catch (Exception e) {
+                        logger.error("Broadcasting has been interrupted" + e.getMessage());
+                        stopBroadCast();
+                    }
+                }
+            });
+        } catch (Exception e) {
+            logger.info("Broadcasting has been interrupted " + e.getMessage());
+            stopBroadCast();
+        }
+    }
+
+    public void stopBroadCast() {
+        BroadCastServer.turnOnBroadcast = false;
+        logger.info("Shutting down BroadCasting");
+        this.sendSocket.close();
```

Comment:
Generally when you close multiple resources you want to guard against exceptions and/or use `finally` clauses so a problem in closing one thing doesn't mean you skip trying to close other things.